### PR TITLE
chore(aws): add mixed regions test for `s3_access_point_public_access_block`

### PR DESCRIPTION
**Pull request by LefterisXefteris**
### Description
This pull request adds a new test case to test_s3_access_point_public_access_block.py that checks the behavior of the s3_access_point_public_access_block check across multiple AWS regions.
Changes

Added test_access_points_mixed_regions function to test S3 access points in US East 1, EU West 1, and AP Southeast 2 regions
The new test case verifies correct handling of different public access block configurations across regions

### Motivation
This test enhances the coverage of the existing test suite by ensuring the check correctly handles:

Access points in multiple regions within a single execution
Varied public access block configurations across different regions
Correct region reporting in the check results

### Testing

Ran the new test case locally using pytest
Verified that all existing tests still pass
Checked that the new test fails appropriately when the check's logic is intentionally broken

### Additional Notes

This test case doesn't require any changes to the main check logic, only adds to the test coverage
Follows the existing pattern of mocking AWS services to avoid actual AWS API calls during testing

### Checklist

 My code follows the style guidelines of this project
 [] I have performed a self-review of my own code
 [] I have commented my code, particularly in hard-to-understand areas
 [] I have made corresponding changes to the documentation (if applicable)
 [] My changes generate no new warnings
 [] I have added tests that prove my fix is effective or that my feature works
 [] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/providers/aws/services/s3/s3_access_point_public_access_block/s3_access_point_public_access_block_test.py [hunk 1]
```diff
@@ -316,3 +316,78 @@ def test_access_points_without_one_public_access_block(self):
...
+                s3control_client,
+            ):
+                check = s3_access_point_public_access_block()
+                result = check.execute()
+
+                assert len(result) == 3
+                assert result[0].status == "PASS"
+                assert result[1].status == "FAIL"
+                assert result[2].status == "FAIL"
+                assert result[2].region == "ap-southeast-2"
```
Comment (pedrooot) [inline tests/providers/aws/services/s3/s3_access_point_public_access_block/s3_access_point_public_access_block_test.py]: Could you add more assertions? (`status_extended`, `resource_id`,  `resource_arn`...) take a look at `test_access_points_without_public_access_block` and try to add all those asserts 🙏🏼

**Review by pedrooot — DISMISSED** [review]
Thanks for taking care of this! 🔝 💯 
Could you explain the difference this new test and `test_access_points_without_one_public_access_block`? (appart from the usage of a new `AccessPoint`)

**Comment by pedrooot** [comment]
Also, review https://docs.prowler.com/projects/prowler-open-source/en/latest/developer-guide/introduction/#contributing-with-your-code-or-fixes-to-prowler to fix the failing pr-tests 🙏🏼

**Comment by MrCloudSec** [comment]
Test has been improved with the needed asserts, thanks @LefterisXefteris !

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**